.github/workflows/build.yml

```diff
@@ -0,0 +1,42 @@
+name: build
+on:
+  # Enable manually triggering this workflow via the API or web UI
+  workflow_dispatch:
+  push:
+    branches:
+      - main
+    tags:
+      - v*
+  pull_request:
+
+defaults:
+  run:
+    shell: bash
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+
+      - uses: actions/setup-go@v4
+        with:
+          go-version: 1.19.x
+
+      - name: Build using the Go compiler and the -race option
+        run: go build -race ./...
```

Comment:
I'm not getting why we need this step. What is testing? We don't use the build process directly by Go. It will use the `xk6` already tested below.

Reply:
My rationale was to include it for the following reasons:
- extensions used in the core (as experimental, for instance) split the package containing the code, `/redis` here, from the extension itself, `register.go`. We might as well verify that the `redis` package successfully builds independently before building the extension itself, as this is [the way k6 uses our module redis](https://github.com/grafana/k6/blob/master/js/jsmodules.go#L34).
- it's also a good opportunity to run it in `-race` detection mode, which I prefer to have, considering we're asynchronous, and I've had mishaps in the past. I can't remember off the top of my head if xk6 can do that same, though. If so, I'd happily move that to the xk6 stage.

If you think it's not worth it, let me know, and I'm happy to drop that step.

Reply:
>We might as well verify that the redis package successfully builds 

Isn't this already covered by `xk6 build` command? I expect they both fail or succeed. Do you see a case where it can build on one command but not on the other?

Reply:
>it's also a good opportunity to run it in -race detection mode

Building without running shouldn't really apply any detection, right?